src/main/java/iterator/RichIterator.java

```diff
@@ -158,7 +229,8 @@ default A min(Comparator<? super A> comparator) {
      * @return a new Iterator.
      */
     default <B> RichIterator<B> map(Function<? super A, ? extends B> f) {
-        throw new NotImplementedException();
+        Iterable<A> iterable = () -> this;
+        return RichIterator.from(StreamSupport.stream(iterable.spliterator(), true).map(f).collect(Collectors.toList()));
```

Comment:
the hole idea of this exercise is to implement it yourself, try avoiding using streams. (I should add that to the README)